### PR TITLE
feat: Inform user about enabling Ent-to-End Identity (WPB-308)

### PR DESCRIPTION
**Pull request by borichellow**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-308" title="WPB-308" target="_blank">EpicWPB-308</a>  [AR] E2EI UI - Alert user when team settings changed
  </td></table>
  <br />
 

# What's new in this PR?

adopted kalium changes to observe the changes in e2eId requirement and show a dialog about it

### Dependencies (Optional)

Needs releases with:

- wireapp/kalium/pull/1832

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

### Attachments (Optional)
Screenshot 2023-07-03 at 17 24 36
Screenshot 2023-07-03 at 17 24 53
Screenshot 2023-07-03 at 17 24 23

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [170](https://10.10.124.134/job/AR-build-pipeline/170/) **failed**.

**Comment by AndroidBob** [comment]
Build [186](https://10.10.124.134/job/AR-build-pipeline/186/) **failed**.

**Comment by AndroidBob** [comment]
Build [188](https://10.10.124.134/job/AR-build-pipeline/188/) **failed**.

**Comment by AndroidBob** [comment]
Build [192](https://10.10.124.134/job/AR-build-pipeline/192/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.3.0(6949505)(6949505)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.3.0(6949505)(6949505)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/FeatureFlagState.kt [hunk 1]
```diff
@@ -30,9 +31,18 @@ data class FeatureFlagState(
...
     enum class SharingRestrictedState {
         NONE, NO_USER, RESTRICTED_IN_TEAM
     }
+
+    data class E2EIdSnooze(val timeLeft: Duration)
+
+    sealed class E2EIdRequired {
+        data class WithGracePeriod(val timeLeft: Duration) : E2EIdRequired()
+        object NoGracePeriod : E2EIdRequired()
+    }
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/FeatureFlagState.kt]: Should both this data classes be in `FeatureFlagState`?

File: app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt [hunk 2]
```diff
@@ -147,6 +149,16 @@ class FeatureFlagNotificationViewModel @Inject constructor(
         }
     }
 
+    private fun setMLSE2EIdRequiredState(userId: UserId) = viewModelScope.launch {
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt]: following some naming suggestions from [this PR](https://github.com/wireapp/kalium/pull/1832) maybe we can remove the `MLS` from the function name?

just: `setE2EIdRequiredState(..` ?

File: app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt [hunk 3]
```diff
@@ -167,4 +179,25 @@ class FeatureFlagNotificationViewModel @Inject constructor(
         }
         featureFlagState = featureFlagState.copy(shouldShowGuestRoomLinkDialog = false)
     }
+
+    fun getE2EICertificate() {
+        // TODO do the magic
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt:184]: I suppose then for this PR it should be only about showing the dialogs, right?

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1091,4 +1091,11 @@
...
+
+    <string name="end_to_end_identity_required_dialog_title">End-to-end identity certificate</string>
+    <string name="end_to_end_identity_required_dialog_text">As of today, your team uses end-to-end identity to make Wire’s usage more secure and practicable. The device verification takes place automatically using a certificate and replaces the previous manual process. This way, you communicate with the highest security standard.\n\nEnter the credentials of your identity provider in the next step to automatically get a verification certificate for this device.</string>
+    <string name="end_to_end_identity_required_dialog_text_no_snooze">Your team now uses end-to-end identity to make Wire’s usage more secure. The device verification takes place automatically using a certificate.\n\nEnter the credentials of your identity provider in the next step to automatically get a verification certificate for this device.</string>
+    <string name="end_to_end_identity_required_dialog_positive_btn">Get certificate</string>
+    <string name="end_to_end_identity_required_dialog_snooze_btn">Remind me later</string>
```
Comment (alexandreferris) [inline app/src/main/res/values/strings.xml]: For naming conventions we can specify the full naming of it, from `btn` to `button`

**Review by alexandreferris — COMMENTED** [review]
Great work! I just have some comments

File: app/src/main/kotlin/com/wire/android/ui/home/FeatureFlagState.kt (hunk 1 above)
Comment (borichellow) [reply app/src/main/kotlin/com/wire/android/ui/home/FeatureFlagState.kt]: Both are related only to FeatureFlagState, so why not?

File: app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt (hunk 3 above)
Comment (borichellow) [reply app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt:184]: yeah, this one is only about informing user, and generating the Certificate will be implemented in other PR and I just put this `TODO` to make it easier to find where to call a usecase

File: app/src/main/kotlin/com/wire/android/ui/home/FeatureFlagState.kt [hunk 5]
```diff
@@ -30,9 +31,18 @@ data class FeatureFlagState(
     val isGuestRoomLinkEnabled: Boolean = true,
     val shouldShowSelfDeletingMessagesDialog: Boolean = false,
     val enforcedTimeoutDuration: SelfDeletionDuration = SelfDeletionDuration.None,
-    val areSelfDeletedMessagesEnabled: Boolean = true
+    val areSelfDeletedMessagesEnabled: Boolean = true,
+    val e2EIRequired: E2EIdRequired? = null,
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/FeatureFlagState.kt]:
```suggestion
    val e2EIdRequired: E2EIdRequired? = null,
```
Missing the `d`? 😅

**Comment by AndroidBob** [comment]
Build [195](https://10.10.124.134/job/AR-build-pipeline/195/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.3.0(6950527)(6950527)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.3.0(6950527)(6950527)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt [hunk 6]
```diff
@@ -147,6 +149,16 @@ class FeatureFlagNotificationViewModel @Inject constructor(
         }
     }
 
+    private fun setE2EIRequiredState(userId: UserId) = viewModelScope.launch {
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt:152]:
```suggestion
    private fun setE2EIdRequiredState(userId: UserId) = viewModelScope.launch {
```
Missing the `d` from `Id` here as well

File: app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt [hunk 7]
```diff
@@ -167,4 +179,25 @@ class FeatureFlagNotificationViewModel @Inject constructor(
         }
         featureFlagState = featureFlagState.copy(shouldShowGuestRoomLinkDialog = false)
     }
+
+    fun getE2EICertificate() {
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt:183]: Missing the `d` from `Id` here as well?

**Review by alexandreferris — COMMENTED** [review]
👌🏻 just missing a character on naming and it should be all good!

**Review by alexandreferris — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [201](https://10.10.124.134/job/AR-build-pipeline/201/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.3.0(6957625)(6957625)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.3.0(6957625)(6957625)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/FeatureFlagState.kt (hunk 5 above)
Comment (mchenani) [reply app/src/main/kotlin/com/wire/android/ui/home/FeatureFlagState.kt]: the 'd' is extra actually :D and I suggest to have it like 'isE2EIRequired'

File: app/src/main/kotlin/com/wire/android/ui/home/FeatureFlagState.kt [hunk 8]
```diff
@@ -30,9 +31,18 @@ data class FeatureFlagState(
...
+    val e2EIdSnoozeInfo: E2EIdSnooze? = null
```
Comment (mchenani) [inline app/src/main/kotlin/com/wire/android/ui/home/FeatureFlagState.kt]: here the d is extra inmho.

File: app/src/main/kotlin/com/wire/android/ui/home/HomeDialogs.kt [hunk 9]
```diff
@@ -131,6 +135,113 @@ fun WelcomeNewUserDialog(
     )
 }
 
+@Composable
+fun E2EIdRequiredDialog(
```
Comment (mchenani) [inline app/src/main/kotlin/com/wire/android/ui/home/HomeDialogs.kt]:
```suggestion
fun E2EIRequiredDialog(
```

**Review by mchenani — APPROVED** [review]
🥳 , some naming suggestions.

File: app/src/main/kotlin/com/wire/android/ui/home/HomeDialogs.kt (hunk 9 above)
Comment (borichellow) [reply app/src/main/kotlin/com/wire/android/ui/home/HomeDialogs.kt]: god damn this `d` :D

**Comment by AndroidBob** [comment]
Build [202](https://10.10.124.134/job/AR-build-pipeline/202/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.3.0(6958577)(6958577)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.3.0(6958577)(6958577)-staging-compat.apk)

**Pull request merged**